neo4j-to-neptune/src/test/java/com/amazonaws/services/neptune/io/Neo4jStreamWriterTest.java

```diff
@@ -0,0 +1,227 @@
+or in the "license" file accompanying this file. This file is distributed
+on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+express or implied. See the License for the specific language governing
+permissions and limitations under the License.
+*/
+
+package com.amazonaws.services.neptune.io;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.Rule;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.junit.Assert.*;
+
+/**
+ * Unit tests for Neo4jStreamWriter class focusing on validation and configuration.
+ * These tests don't require a Neo4j database connection.
+ */
+public class Neo4jStreamWriterTest {
+
+    @Rule
+    public TemporaryFolder tempFolder = new TemporaryFolder();
+
+    private Directories directories;
+
+    @Before
+    public void setUp() throws IOException {
+        // Create a temporary directory for testing
+        File tempDir = tempFolder.newFolder("test-output");
+        directories = Directories.createFor(tempDir);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConstructorWithNullUri() {
+        new Neo4jStreamWriter(null, "neo4j", "password", directories);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConstructorWithEmptyUri() {
+        new Neo4jStreamWriter("", "neo4j", "password", directories);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConstructorWithWhitespaceUri() {
+        new Neo4jStreamWriter("   ", "neo4j", "password", directories);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConstructorWithNullUsername() {
+        new Neo4jStreamWriter("bolt://localhost:7687", null, "password", directories);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConstructorWithNullPassword() {
+        new Neo4jStreamWriter("bolt://localhost:7687", "neo4j", null, directories);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConstructorWithNullDirectories() {
+        new Neo4jStreamWriter("bolt://localhost:7687", "neo4j", "password", null);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConstructorWithNullConfig() {
+        new Neo4jStreamWriter("bolt://localhost:7687", "neo4j", "password", directories, null);
+    }
+
+    @Test
+    public void testConstructorValidationPasses() {
+        // This will fail with connection error, but validation should pass
+        try  {
+            new Neo4jStreamWriter("bolt://localhost:7687", "neo4j", "password", directories);
+            fail("Should have thrown RuntimeException due to connection failure");
+        } catch (RuntimeException e) {
+            // Expected - connection will fail, but validation passed
+            assertTrue("Should fail due to connection, not validation", 
```

Comment:
Would be good to assert that `e` is not an `IllegalArgumentException` thrown from the validation as `IllegalArgumentException` is a subclass of `RuntimeException`